src/rules/no-confusing-set-timeout.ts

```diff
@@ -0,0 +1,81 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import { createRule, getNodeName, parseJestFnCall } from './utils';
+
+interface ErrorType {
+  messageId: 'globalSetTimeout' | 'multipleSetTimeouts' | 'orderSetTimeout';
+  node: TSESTree.Node;
+}
+
+function isJestSetTimeout(node: TSESTree.Node) {
+  return getNodeName(node) === 'jest.setTimeout';
+}
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Disallow using confusing setTimeout in test',
+      recommended: false,
+    },
+    messages: {
+      globalSetTimeout: '`jest.setTimeout` should be call in `global` scope.',
+      multipleSetTimeouts:
+        'Do not call `jest.setTimeout` multiple times, as only the last call will have an effect.',
+      orderSetTimeout:
+        '`jest.setTimeout` should be placed before any other jest methods',
+    },
+    type: 'problem',
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    const errors: ErrorType[] = [];
+    let callJestTimeout = 0;
+    let nonJestTimeout = 0;
+
+    return {
+      CallExpression(node) {
```

Comment:
My gut is saying we should be able to make this simpler by focusing on the conditions for each message we're checking for kind of like bailing early, rather than tracking counts - i.e.
  * if we're not in the global scope, then any use of `setTimeout` should trigger `globalSetTimeout`
  * if we've already see a usage of `jest.setTimeout`, then any other usage we come across should trigger `multipleSetTimeout`
  * if we see a `setTimeout` after seeing any other jest function, then that should trigger `orderSetTimeout`

Could you explore this and see if it is possible?